packages/blade/src/components/Tabs/__tests__/Tabs.web.test.tsx

```diff
@@ -0,0 +1,442 @@
+import userEvent from '@testing-library/user-event';
+import { fireEvent, screen } from '@testing-library/react';
+import React from 'react';
+import { Tabs, TabList, TabItem, TabPanel } from '../';
+import renderWithTheme from '~utils/testing/renderWithTheme.web';
+import { Text } from '~components/Typography';
+import { Box } from '~components/Box';
+import { Button } from '~components/Button';
+import assertAccessible from '~utils/testing/assertAccessible.web';
+
+const queryTabPanelByText = (text: string): HTMLElement | null => {
+  return (
+    screen.queryAllByRole('tabpanel', { hidden: true }).find((el) => el.textContent === text) ??
+    null
+  );
+};
+
+describe('Tabs', () => {
+  it('should render', () => {
+    const { baseElement, queryAllByRole } = renderWithTheme(
+      <Tabs defaultValue="refunds">
+        <TabList>
+          <TabItem value="payments">Payments</TabItem>
+          <TabItem value="refunds">Refunds</TabItem>
+        </TabList>
+
+        <TabPanel value="payments">
+          <Text>Payments</Text>
+        </TabPanel>
+        <TabPanel value="refunds">
+          <Text>Refunds</Text>
+        </TabPanel>
+      </Tabs>,
+    );
+
+    expect(baseElement).toMatchSnapshot();
+
+    // `queryByRole` is failing but `queryAllByRole` is working
+    // I think because even with `hidden: true` the `name` will try to assert the `accessibleName`
+    // Which never gets computed because `aria-hidden` is set
+    // To work around this, I've added a custom `queryTabPanelByText` function
+    // expect(queryByRole('Payments', idden: true })).toBeInTheDocument();
+
+    // Expect all tab panles to be in the dom
+    expect(queryAllByRole('tabpanel', { hidden: true })).toHaveLength(2);
+    expect(queryTabPanelByText('Payments')).toBeInTheDocument();
+    expect(queryTabPanelByText('Refunds')).toBeInTheDocument();
+  });
+
+  it('should render with vertical orientation', () => {
```

Comment:
the orientation is not supported on mobile?

Reply:
No not really. While you can do it, it's not a valid variant since there not enough space for it to be any useful. + We already prevent this variant from figma side.